### PR TITLE
Core/Packet: Character

### PR DESCRIPTION
**Pull request by IntelligentQuantum**
## Changes Proposed:
-  Class Base
-  

## Issues Addressed:

- Closes 

## SOURCE:

TC
## Tests Performed:

- 
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Handlers/CharacterHandler.cpp [hunk 1]
```diff
@@ -1194,9 +1195,9 @@ void WorldSession::HandlePlayerLoginToCharInWorld(Player* pCurrChar)
 
                 _mask[i] = uint32(1) << (eff - (32 * i));
                 int32 val = 0;
-                for (SpellModList::const_iterator itr = spellMods.begin(); itr != spellMods.end(); ++itr)
-                    if ((*itr)->type == modType && (*itr)->mask & _mask)
-                        val += (*itr)->value;
+                for (auto spellMod : spellMods)
```
Comment (Winfidonarleyan) [inline src/server/game/Handlers/CharacterHandler.cpp:1198]: auto const&

**Review by Winfidonarleyan — APPROVED** [review]

**Comment by temperrr** [comment]
How can one test this PR?

**Comment by IntelligentQuantum** [comment]
> How can one test this PR? What should we look for?

Logout/ Cancel Logout
Played Time
Showing Helm 
Showing Cloak

**Comment by temperrr** [comment]
Tested

- [x] no compilation errors
- [x] can show/hide cloak
- [x] can show/hide helm
- [x] can logout
- [x] /played works correctly

**Pull request merged**